sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_models.py

```diff
@@ -101,21 +104,25 @@ def id(self):
 
     @property
     def name(self):
-        # type: () -> str
+        # type: () -> Optional[str]
```

Comment:
I worry this encourages users to write defensive code they will never need. We tell users this property could be `None` but they don't need to consider that, because it only occurs in our internal usage of this class, never through the public API (modulo the influence of bugs we would want to know about).